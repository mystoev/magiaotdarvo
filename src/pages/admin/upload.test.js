import React from 'react';
import { fireEvent, waitFor } from '@testing-library/react';
import { renderWithTheme } from '../../../test/utils';

import Upload from './upload';

describe('<Upload /> tests', () => {
  it('should render', () => {
    const { container } = renderWithTheme(<Upload />);

    expect(container).toMatchSnapshot();
  });

  it('should upload', async () => {
    const file = new File(['some content'], 'file1.jpg', { type: 'image/jpg' });
    const onUploadMock = jest.fn();
    const { queryByTestId } = renderWithTheme(<Upload onUpload={onUploadMock} />);

    fireEvent.change(queryByTestId('file-input'), { target: { files: [file] } });

    await waitFor(() => {
      expect(onUploadMock).toHaveBeenCalledWith([
        {
          data: 'data:image/jpg;base64,c29tZSBjb250ZW50',
          name: 'file1.jpg'
        }
      ]);
    });
  });
});
