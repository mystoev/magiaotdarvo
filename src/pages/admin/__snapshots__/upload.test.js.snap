// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Upload /> tests should render 1`] = `
.c0 {
  display: inline-block;
  background: #ffffff;
  position: relative;
  height: 200px;
}

.c0 img {
  height: 100%;
  width: auto;
}

.c0 input {
  display: block;
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: 0;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <img
      src="[object Object]"
    />
    <input
      data-testid="file-input"
      multiple=""
      title="Добави снимка"
      type="file"
    />
  </div>
</div>
`;
